### ISSUE TITLE
Dropdown: Wrong behavior of `home`, `end`, `left arrow` and `right arrow` while `editable`

### ISSUE THREAD

**Issue by DiamondeX** [primefaces/primereact#8072]
### Describe the bug

**Accessibility** section describes:

> `right arrow`: If the dropdown is editable, _removes the visual focus from the current option_ and moves input cursor to one character left.
> `left arrow`: If the dropdown is editable, _removes the visual focus from the current option_ and moves input cursor to one character right.
> `home`: _If the dropdown is editable, moves input cursor at the end_, if not then moves focus to the first option.
> `end`: _If the dropdown is editable, moves input cursor at the beginning_, if not then moves focus to the last option.

In reality described features don't work. It's easy to verify if you will try "editable" example of the Dropdown docs (https://primereact.org/dropdown/#editable)

### Reproducer

https://stackblitz.com/edit/juvrshuk

### System Information

```Shell
System:
    OS: Windows 11 10.0.26100
    CPU: (32) x64 Intel(R) Core(TM) i9-14900HX
    Memory: 4.09 GB / 31.71 GB
  Binaries:
    Node: 22.14.0 - C:\Program Files\nodejs\node.EXE
    npm: 10.9.2 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Chromium (133.0.3065.82)
  npmPackages:
    primereact: ^10.9.6 => 10.9.6
    react: ^19.0.0 => 19.0.0
    tailwindcss: ^4.0.17 => 4.0.17
```

### Steps to reproduce the behavior

1. go to https://primereact.org/dropdown/#editable
2. click the input field of the editable dropdown or focus it using keyboard
3. input some text
4. press "home" (or "end") key: **_first (or last) option in dropdown list will become focused._** 

Now no key of the mentioned above will focus input field again as expected.
It works the same without text in input field.
It works the same with "Shift" key.

### Expected behavior

At step (4) you should have:  press "home" (or "end") key: input cursor should move to the start (or end) of the input field. 

In editable mode:
- "home" and "end" keys should move focus from dropdown list to input field and not vise versa.
- "left arrow" and "right arrow" keys should move focus from dropdown list to input field.
- using "Shift" key shouldn't change anything: it should act the same as in normal input field

The only behavior that is ok and should stay is that left and right arrow keys don't move focus to dropdown list.
